scripts/pylib/twister/twisterlib.py

```diff
@@ -681,16 +681,19 @@ def monitor_serial(self, ser, halt_fileno, harness):
             # is available yet.
             if serial_line:
                 sl = serial_line.decode('utf-8', 'ignore').lstrip()
-                logger.debug("DEVICE: {0}".format(sl.rstrip()))
+                try:
+                    logger.debug("DEVICE: {0}".format(sl.rstrip()))
```

Comment:
will this report UnicodeEncodeError ? seems like all invalid chars are already handled by line above "sl = serial_line.decode('utf-8', 'ignore').lstrip()", the output sl is normal, it won't meet any encode/decode error.

Reply:
Hi, Peng, no, it doesn't work and still caught an UnicodeEncodeError exception on this. I think this message might comes from grub stage.

Reply:
And this only happened at that certain moment once, the rest of the message doesn't lead to such an error.